### PR TITLE
Feature/rules

### PR DESCRIPTION
**Pull request by Rutu2k**
Profile component and container setup is done.(Profile Component will be combination of 1.Rules Component and 2. Candidate Form)
Added Rules Component with API integration(Assumed Dummy API) with redux store and redux saga also handled the errors. 
Done with Unit testing of Rules Reducer of Rules Component.
Please Review and suggest changes if any.

### CODE REVIEW

File: src/actions/rulesAction.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import RULES_REDUCER from "../constants/actionConstants";
+
+export const rulesRequest = () => {
```
Comment (vaibhavchalse99) [inline src/actions/rulesAction.js:3]: Single statement can be returned directly

export const rulesRequest = ()=>({ type: RULES_REDUCER.RULES_REQUEST})

Comment (Rutu2k) [reply src/actions/rulesAction.js:3]: sure, Thanks

File: src/actions/rulesAction.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    return {
+        type: RULES_REDUCER.RULES_REQUEST,
+    }
+}
+export const rulesRequestFailed = (data) => {
+    return {
+        type: RULES_REDUCER.RULES_REQUEST_FAILED,
+        payload: {requestError: data}
```
Comment (vaibhavchalse99) [inline src/actions/rulesAction.js:11]: requestError: data
parameter should be rename with the proper name

File: src/actions/rulesAction.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    }
+}
+const rulesAction = (data) => {
+    return {
+        type: RULES_REDUCER.SET_RULES, 
+        payload: {userlist: data}
+    }
+}
+export default rulesAction;
```
Comment (vaibhavchalse99) [inline src/actions/rulesAction.js:20]: why default export?

File: src/actions/rulesAction.js (hunk 2 above)
Comment (Rutu2k) [reply src/actions/rulesAction.js:11]: okay

File: src/apis/rulesAPI.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import RULES_LIST_URL from "constants/appConstants"
+const axios = require('axios').default;
```
Comment (vaibhavchalse99) [inline src/apis/rulesAPI.js:2]: why require?

File: src/actions/rulesAction.js (hunk 3 above)
Comment (Rutu2k) [reply src/actions/rulesAction.js:20]: You are right ..its redundant

File: src/store.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
+// import rootReducer from "reducers/rootReducer";
+import RulesReducer from "reducers/RulesReducer";
 import rootSagas from "sagas/rootSagas";
 
 const sagaMiddleware = createSagaMiddleware();
 
-export const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
+export const store = createStore(RulesReducer,applyMiddleware(sagaMiddleware));
 
 sagaMiddleware.run(rootSagas);
```
Comment (vaibhavchalse99) [inline src/store.js:12]: you don't have to do anything with this file. 
remove these changes this is the setup of store.
ping me for the same

**Review by vaibhavchalse99 — COMMENTED** [review]
do the changes mentioned. 
ping me if any difficulty

File: src/store.js (hunk 5 above)
Comment (Rutu2k) [reply src/store.js:12]: okay

File: src/actions/rulesAction.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { RULES } from 'constants/actionConstants';
+
+export const rulesRequest = () => ({ type: RULES.DETAIL_REQUEST });
+
+export const rulesRequestFailed = (requestError) => ({
+  type: RULES.SET_ERROR_MESSAGE,
+  payload: { requestError },
```
Comment (Suhas010) [inline src/actions/rulesAction.js:7]: is there any specific reason you are making payload an object?

File: src/containers/RulesContainer.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+
+import RulesComponent from 'components/RulesComponent';
+import { rulesRequest } from '../actions/rulesAction';
+
+const RulesContainer = () => {
+  const dispatch = useDispatch();
+  const result = useSelector((state) => state.RulesReducer);
+
+  let data = result.userlist.str;
```
Comment (Suhas010) [inline src/containers/RulesContainer.jsx:11]: destructure?

File: src/containers/RulesContainer.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+  const { userlist : { str : data }, requestError } = result;
+  let flag = false;
+
+  useEffect(() => {
+    dispatch(rulesRequest());
+  }, [dispatch]);
+
+  if (requestError) {
+    flag = true;
```
Comment (shailendrakanherkar18) [inline src/containers/RulesContainer.jsx:19]: why to have separate flag cboolean flag

**Review by shailendrakanherkar18 — APPROVED** [review]

**Pull request merged**